pennylane/devices/default_mixed.py

```diff
@@ -905,3 +977,80 @@ def execute(
         execution_config: Optional[ExecutionConfig] = None,
     ) -> None:
         raise NotImplementedError
+
+    def _setup_execution_config(self, execution_config: ExecutionConfig) -> ExecutionConfig:
+        """This is a private helper for ``preprocess`` that sets up the execution config.
+
+        Args:
+            execution_config (ExecutionConfig): an unprocessed execution config.
+
+        Returns:
+            ExecutionConfig: a preprocessed execution config.
+        """
+        updated_values = {}
+        for option in execution_config.device_options:
+            if option not in self._device_options:
+                raise qml.DeviceError(f"device option {option} not present on {self}")
+
+        if execution_config.gradient_method == "best":
+            updated_values["gradient_method"] = "backprop"
+        updated_values["use_device_gradient"] = execution_config.gradient_method in {
+            "backprop",
+            "best",
+        }
+        updated_values["grad_on_execution"] = False
+        updated_values["device_options"] = dict(execution_config.device_options)  # copy
+
+        for option in self._device_options:
+            if option not in updated_values["device_options"]:
+                updated_values["device_options"][option] = getattr(self, f"_{option}")
+        return replace(execution_config, **updated_values)
+
+    @debug_logger
+    def preprocess(
+        self,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
```

Comment:
```suggestion
        execution_config: ExecutionConfig = None,
```
we're trying to move away from using this global mutable object as the default execution config. Instead just make it `None` and in the first line of your preprocess initialize it to the default execution config:
```
execution_config = execution_config or ExecutionConfig()
```